File: python/muse/util/random_factory.py
import secrets


def gen_int_n(min: int, max: int) -> int:
    return min + secrets.randbelow(max - min + 1)


def gen_int() -> int:
    return gen_int_n(1, 2_147_483_647)


def gen_even() -> int:
    return gen_int() >> 1 << 1


def gen_odd() -> int:
    return gen_int() | 1

File: rust/muse/src/util/random_factory.rs
use rand::{thread_rng, Rng};

pub fn gen_int_n(min: i32, max: i32) -> i32 {
    let mut rng = thread_rng();
    rng.gen_range(min..=max)
}

pub fn gen_int() -> i32 {
    gen_int_n(1, 2_147_483_647)
}

pub fn gen_even() -> i32 {
    gen_int() >> 1 << 1
}

pub fn gen_odd() -> i32 {
    gen_int() | 1
}

#[cfg(test)]
mod tests;
